loader_hub/googledrive/base.py

```diff
@@ -0,0 +1,220 @@
+"""Google Drive files reader."""
+
+from pathlib import Path
+from typing import Dict, List, Optional, Any, Union, Callable
+import os
+import logging
+import tempfile
+
+from gpt_index import download_loader
+from gpt_index.readers.base import BaseReader
+from gpt_index.readers.schema.base import Document
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+
+from pydrive.auth import GoogleAuth
+from pydrive.drive import GoogleDrive
+from googleapiclient.discovery import build
+
+logger = logging.getLogger(__name__)
+
+# Scope for reading and downloading google drive files
+SCOPES = ["https://www.googleapis.com/auth/drive.readonly"]
+
+
+class GoogleDriveReader(BaseReader):
+    """Google drive reader."""
+
+    def __init__(self,
+                 credentials_path: str = "credentials.json",
+                 client_secrets_path: str = "client_secrets.json",
+                 token_path: str = "token.json") -> None:
+        """Initialize with parameters."""
+        self.credentials_path = credentials_path
+        self.client_secrets_path = client_secrets_path
+        self.token_path = token_path
+
+    def _get_credentials(self) -> Any:
```

Comment:
I think there's something broken with this function. It's trying to refresh and make me login every single time I run it.

Reply:
that's because of following part
`gauth = GoogleAuth()
  gauth.LocalWebserverAuth()
  drive = GoogleDrive(gauth)`